test/e2e/specs/editor/blocks/image.spec.js

```diff
@@ -729,6 +729,190 @@ test.describe( 'Image', () => {
+		await expect( image ).toHaveAttribute( 'src', new RegExp( filename ) );
+
+		await editor.openDocumentSettingsSidebar();
+	} );
+
+	test.afterEach( async ( { requestUtils, admin, page } ) => {
+		await requestUtils.deleteAllMedia();
+
+		await admin.visitAdminPage(
+			'/admin.php',
+			'page=gutenberg-experiments'
+		);
+
+		await page
+			.locator( `#gutenberg-interactivity-api-core-blocks` )
+			.setChecked( false );
+
+		await page.locator( `input[name="submit"]` ).click();
+
+		await page.waitForLoadState();
+	} );
+
+	test( 'should toggle "lightbox" in saved attributes', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.getByRole( 'button', { name: 'Advanced' } ).click();
+		await page.getByLabel( 'Behaviors' ).selectOption( 'Lightbox' );
+		const testForTrue = new RegExp(
+			`<!-- wp:image {"id":(\\d+),"sizeSlug":"full","linkDestination":"none","behaviors":{"lightbox":true}} -->
+<figure class="wp-block-image size-full"><img src="[^"]+\\/${ filename }\\.png" alt="" class="wp-image-\\1"/></figure>
+<!-- \\/wp:image -->`
+		);
+		expect( await editor.getEditedPostContent() ).toMatch( testForTrue );
+
+		await page.getByLabel( 'Behaviors' ).selectOption( '' );
+		const testForFalse = new RegExp(
+			`<!-- wp:image {"id":(\\d+),"sizeSlug":"full","linkDestination":"none","behaviors":{"lightbox":false}} -->
+<figure class="wp-block-image size-full"><img src="[^"]+\\/${ filename }\\.png" alt="" class="wp-image-\\1"/></figure>
+<!-- \\/wp:image -->`
+		);
+		expect( await editor.getEditedPostContent() ).toMatch( testForFalse );
+	} );
+
+	test( 'should open and close the image in a lightbox using the mouse', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.getByRole( 'button', { name: 'Advanced' } ).click();
+		await page.getByLabel( 'Behaviors' ).selectOption( 'Lightbox' );
+
+		postId = await editor.publishPost();
+		await page.goto( `/?p=${ postId }` );
+
+		const lightbox = page.locator( '.wp-lightbox-overlay' );
+		await expect( lightbox ).toBeHidden();
+
+		const image = lightbox.locator( 'img' );
+		await expect( image ).toHaveAttribute( 'src', new RegExp( filename ) );
+
+		await page
+			.getByRole( 'button', { name: 'Open image lightbox' } )
+			.click();
+
+		await expect( lightbox ).toBeVisible();
+
+		const closeButton = page.getByRole( 'button', {
+			name: 'Close lightbox',
+		} );
+		await closeButton.click();
+
+		await expect( lightbox ).toBeHidden();
+	} );
+
+	test.describe( 'keyboard navigation', () => {
+		let openLightboxButton;
+		let lightbox;
+		let closeButton;
+
+		test.beforeEach( async ( { page, editor } ) => {
+			await page.getByRole( 'button', { name: 'Advanced' } ).click();
+			await page.getByLabel( 'Behaviors' ).selectOption( 'Lightbox' );
+
+			postId = await editor.publishPost();
+			await page.goto( `/?p=${ postId }` );
+
+			openLightboxButton = page.getByRole( 'button', {
+				name: 'Open image lightbox',
+			} );
+			lightbox = page.locator( '.wp-lightbox-overlay' );
```

Comment:
Could this be refactored into `getByRole`? If it's not accessible, could we add a `data-testid`? If it's not desirable, which I understand for frontend code, could we add a simple comment above here for readers?

Reply:
Done 👍